ufl/core/interpolate.py

```diff
@@ -36,33 +38,35 @@ def __init__(self, expr, v):
         from ufl.algorithms import extract_arguments
 
         expr = as_ufl(expr)
+
         if isinstance(expr, BaseForm):
-            raise ValueError("Expecting the first argument to be primal.")
-        expr_args = extract_arguments(expr)
+            *expr_args, source_coarg = expr.arguments()
```

Comment:
As @dham pointed out at a meeting, we should drop the first argument. Suppose `expr`
 is a forward 2-form `Interpolate(TrialFunction(V1), TestFunction(V2.dual()))`, should its arguments be like this?
```suggestion
            source_coarg, *expr_args = expr.arguments()
```
The other case is the adjoint 2-form Interpolate. We should not worry about it because it maps duals to duals (therefore not a primal expression).